website/upgrade-guides/0-13.html.markdown

```diff
@@ -0,0 +1,367 @@
+currently using a version of Terraform prior to v0.12 please upgrade through
+the latest minor releases of all of the intermediate versions first, reviewing
+the previous upgrade guides for any considerations that may be relevant to you.
+
+In particular, Terraform v0.13 no longer includes the `terraform 0.12upgrade`
+command for automatically migrating module source code from v0.11 to v0.12
+syntax. If your modules are written for v0.11 and earlier you may need to
+upgrade their syntax using the latest minor release of Terraform v0.12 before
+using Terraform v0.13.
+
+-> If you run into any problems during upgrading that are not solved by the
+content below, please feel free to start a topic in
+[The Terraform community forum](https://discuss.hashicorp.com/c/terraform-core),
+describing the problem you've encountered in enough detail that other readers
+may be able to reproduce it and offer advice.
+
+Upgrade guide sections:
+
+* [Explicit Provider Source Locations](#explicit-provider-source-locations)
+* [New Filesystem Layout for Local Copies of Providers](#new-filesystem-layout-for-local-copies-of-providers)
+  * [Special considerations for in-house providers](#in-house-providers)
+* [Destroy-time provisioners may not refer to other resources](#destroy-time-provisioners-may-not-refer-to-other-resources)
+* [Data resource reads can no longer be disabled by `-refresh=false`](#data-resource-reads-can-no-longer-be-disabled-by--refresh-false)
+
+## Explicit Provider Source Locations
+
+Prior versions of Terraform have supported automatic provider installation only
+for providers packaged and distributed by HashiCorp. Providers built by the
+community have previously required manual installation by extracting their
+distribution packages into specific local filesystem locations.
+
+Terraform v0.13 introduces a new hierarchical namespace for providers that
+allows specifying both HashiCorp-maintained and community-maintained providers
+as dependencies of a module, with community providers distributed from other
+namespaces on [Terraform Registry](https://registry.terraform.io/) from a
+third-party provider registry.
+
+In order to establish the hierarchical namespace, Terraform now requires
+explicit source information for any providers that are not HashiCorp-maintained,
+using a new syntax in the `required_providers` nested block inside the
+`terraform` configuration block:
+
+```hcl
+terraform {
+  required_providers {
+    azurerm = {
+      # The "hashicorp" namespace is the new home for the HashiCorp-maintained
+      # provider plugins.
+      #
+      # source is not required for the hashicorp/* namespace as a measure of
+      # backward compatibility for commonly-used providers, but recommended for
+      # explicitness.
+      source  = "hashicorp/azurerm"
+      version = "~> 2.12"
+    }
+    datadog = {
+      # source is required for providers in other namespaces, to avoid ambiguity.
+      source  = "terraform-providers/datadog"
```

Comment:
Suggestion: We might want to switch this to an example provider who has moved out of `terraform-providers` now that has started happening:
> "newrelic/newrelic"

This is a matter of opinion, it's fine if you prefer it as is!

Reply:
I have updated it in this way.

The `terraform 0.13upgrade` command is designed to select what Terraform 0.12 would've installed, so it will actually still use `terraform-providers/newrelic` (the `newrelic/newrelic` provider only has new releases after 2.1.1) but I think it's better to show a third-party namespace in this _particular_ example because this is here mainly just to highlight what this new feature is all about, not as an example for the user to copy. (The "real" example is generated by the `terraform 0.13upgrade` command.)